src/Numerics/Interpolation/LinearSpline.cs

```diff
@@ -130,6 +130,24 @@ public static LinearSpline Interpolate(IEnumerable<double> x, IEnumerable<double
         /// <returns>Interpolated value x(t).</returns>
         public double Interpolate(double t)
         {
+            return Interpolate(t, true);
+        }
+
+        /// <summary>
+        /// Interpolate at point t.
+        /// </summary>
+        /// <param name="t">Point t to interpolate at.</param>
+        /// <param name="allowExtrapolate">set to fals if extrapolate should not be allowed</param>
+        /// <returns>Interpolated value x(t).</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when t is outside range and would result in Extrapolation!</exception>
+        public double Interpolate(double t, bool allowExtrapolate)
```

Comment:
That all the interpolators extrapolate by default is possibly surprising behaviour, especially given that is not documented (as far as I can tell).
But I wonder if `InterpolateWithoutExtrapolation` or the proposed implementation (with `allowExtrapolate`) will resolve this in general.
Presumably users of the other splines would also want this method which would mean adding this method to all classes and to the interface.

Perhaps a cleaner implementation would be to create a decorator class called `NonExtrapolatingSpline` (i.e., implements `IInterpolation` and wraps `IInterpolation`), that handles this in general for all the interpolators.
Also, maybe doc comments could be added to each of the interpolators indicating that they extrapolate by default, but that this new class exists if anyone wants to opt out.